src/api/BundlePush/CmciConfig.ts

```diff
@@ -0,0 +1,88 @@
+/*
+* This program and the accompanying materials are made available under the terms of the
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright IBM Corp, 2019
+*
+*/
+
+"use strict";
+
+import { IHandlerParameters, IProfile } from "@zowe/imperative";
+
+
+/**
+ * Class to implement CMCI configuration.
+ *
+ * @export
+ * @class CmciConfig
+ */
+export class CmciConfig {
+
+  /**
+   * Merge command line overrides together with the optional cmciProfile into a
+   * composite whole
+   */
+  public static mergeProfile(cmciProfile: IProfile, params: IHandlerParameters) {
+
+    if (params.arguments.ch !== undefined) {
+      cmciProfile.host = params.arguments.ch;
+      cmciProfile.H = params.arguments.ch;
+    }
+
+    if (params.arguments.cpo !== undefined) {
+      cmciProfile.port = params.arguments.cpo;
+      cmciProfile.P = params.arguments.cpo;
+    }
+
+    if (params.arguments.cu !== undefined) {
+      cmciProfile.user = params.arguments.cu;
+      cmciProfile.u = params.arguments.cu;
+    }
+
+    if (params.arguments.cpw !== undefined) {
+      cmciProfile.password = params.arguments.cpw;
+      cmciProfile.pw = params.arguments.cpw;
+    }
+
+    if (params.arguments.cru !== undefined) {
+      cmciProfile.rejectUnauthorized = params.arguments.cru;
+      cmciProfile.ru = params.arguments.cru;
+    }
+
+    if (params.arguments.cpr !== undefined) {
+      cmciProfile.protocol = params.arguments.cpr;
+      cmciProfile.u = params.arguments.cpr;
```

Comment:
I believe this should be `cmciProfile.o`

Reply:
Ah yes, well spotted.